### ISSUE TITLE
Solid border around context menu

### ISSUE THREAD

**Issue by Mark42XLII** [black7375/Firefox-UI-Fix#303]
**Describe the bug**

There's a solid, 4px thick border around the context menu.

**Expected behavior**

There should be no border. Technically, the area the border occupies should have 100% transparency, like every other menu and like before the bug was introduced.

**Screenshots**

Dark theme.
![dark](https://user-images.githubusercontent.com/23386935/148655418-7bf22550-578c-4f34-b6f3-deb9d152b05b.png)
Light theme.
![light](https://user-images.githubusercontent.com/23386935/148655420-cefe0811-ed78-43c5-b2df-0233b26fa538.png)

**Environment:**

 - Distribution
   - [x] [Original Lepton](https://github.com/black7375/Firefox-UI-Fix)
   - [x] [Lepton's photon style](https://github.com/black7375/Firefox-UI-Fix/tree/photon-style)
   - [x] [Lepton's proton style](https://github.com/black7375/Firefox-UI-Fix/tree/proton-style)
 - Firefox Version: 
 - OS:
   - [x] Linux
   - [ ] Windows
   - [ ] Mac
   - [ ] Other: 
 - Theme:
   - [x] Light
   - [x] Dark
   - [x] Alpenglow (System: Light)
   - [x] Alpenglow (System: Dark)
   - [x] System Default
     - GTK Theme Name (Linux only): Adwaita, Adwaita-dark, all Materia themes `/bin/bash -c 'Materia{,-dark,-light}{,-compact}'`
   - [x] Other: (e.g. [Photon Colors](https://addons.mozilla.org/en-US/firefox/addon/photon-colors/), [Google Chrome Dark](https://addons.mozilla.org/en-US/firefox/addon/google-chrome-dark/), [Fairytale Of Nature](https://addons.mozilla.org/en-US/firefox/addon/fairytale-of-nature/))

**Additional context**

Affects every theme I could possibly test. It looks like the border takes on the theme's primary color. I'm aware Firefox started showing this issue on v95 and apparently got it fixed somewhere along v96 (#259), but this started to reappear on Nightly v97 at least a month ago (could be even earlier) specifically after commit 7eae852013b66acbfd38f85786d75f16fa188208.

**Comment by smowtenshi** [comment]
Which version/commit of the theme you're running? Did you test both `master` and `dev` branches? What about your Firefox version?

I'm currently on commit dcb1e6f5ece92730aec17ef992e14a212d9ee503 with slight modifications and since first release of Firefox Nightly 97, I no longer face this issue.

**Comment by Mark42XLII** [comment]
It happens with both branches and their `photon-style`/`proton-style` counterparts.

I normally run Developer Edition or Beta - when I noticed the the original issue was closed, the fixed version hadn't landed yet on those editions. After installing Nightly 97.0a1 (2021-12-11) to check it out, the bug was already present.

Since then, I managed to confirm the issue at least on Debian bookworm, Ubuntu 20.10 and Arch, and it has never once disappeared as Nightly updated. As I said before, 7eae852013b66acbfd38f85786d75f16fa188208 is the first bad commit for me. Anything before that works.

**Comment by smowtenshi** [comment]
I can't reproduce it at all.

**Comment by black7375** [comment]
Normally this commit should apply and be resolved. https://github.com/black7375/Firefox-UI-Fix/commit/0bb237a39e2b31629c28ef823a093269ae75291e

The following commit has no side effect on the context menu. https://github.com/black7375/Firefox-UI-Fix/commit/7eae852013b66acbfd38f85786d75f16fa188208

Please try `git pull` to update or reinstall to the latest version.

**Comment by Mark42XLII** [comment]
Figured it out: the environment variable `XDG_CURRENT_DESKTOP` was unset.

It looks like since Firefox 97, #259 was fixed, but transparency on the context menu when the desktop environment is unknown was disabled... except when there's `-moz-proton`, which breaks Lepton around the tab and URL bar areas (#276) and reverts them back to stock proton - not the context menu, though. I ran `git bisect` while only paying attention to the menu, which still looked fine-ish on this broken state.

Either way, even *if* this is a bug, it's upstream, but I doubt anyone will care about supporting undefined behavior.
